core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -126,23 +197,681 @@ public Response getInfo(@QueryParam(QUERY_RAW_CONFIGURATION) final Boolean rawCo
+        diskFreeCapacity = "UNKNOWN";
+      }
+      long capacityBytes = mountInfo.getUfsCapacityBytes();
+      long usedBytes = mountInfo.getUfsUsedBytes();
+      long freeBytes = -1;
+      if (capacityBytes >= 0 && usedBytes >= 0 && capacityBytes >= usedBytes) {
+        freeBytes = capacityBytes - usedBytes;
+      }
+
+      String totalSpace = "UNKNOWN";
+      if (capacityBytes >= 0) {
+        totalSpace = FormatUtils.getSizeFromBytes(capacityBytes);
+      }
+      diskCapacity = totalSpace;
+
+      String usedSpace = "UNKNOWN";
+      if (usedBytes >= 0) {
+        usedSpace = FormatUtils.getSizeFromBytes(usedBytes);
+      }
+      diskUsedCapacity = usedSpace;
+
+      String freeSpace = "UNKNOWN";
+      if (freeBytes >= 0) {
+        freeSpace = FormatUtils.getSizeFromBytes(freeBytes);
+      }
+      diskFreeCapacity = freeSpace;
+
+      StorageTierAssoc globalStorageTierAssoc = mBlockMaster.getGlobalStorageTierAssoc();
+      List<StorageTierInfo> infos = new ArrayList<>();
+      Map<String, Long> totalBytesOnTiers = mBlockMaster.getTotalBytesOnTiers();
+      Map<String, Long> usedBytesOnTiers = mBlockMaster.getUsedBytesOnTiers();
+
+      for (int ordinal = 0; ordinal < globalStorageTierAssoc.size(); ordinal++) {
+        String tierAlias = globalStorageTierAssoc.getAlias(ordinal);
+        if (totalBytesOnTiers.containsKey(tierAlias) && totalBytesOnTiers.get(tierAlias) > 0) {
+          StorageTierInfo info = new StorageTierInfo(tierAlias, totalBytesOnTiers.get(tierAlias),
+              usedBytesOnTiers.get(tierAlias));
+          infos.add(info);
+        }
+      }
+
+      return new WebUIOverview().setCapacity(getCapacityInternal()).setConfigCheckErrorNum(
+          report.getConfigErrors().values().stream().mapToInt(List::size).sum())
+          .setConfigCheckErrors(report.getConfigErrors())
+          .setConfigCheckStatus(report.getConfigStatus())
+          .setConfigCheckWarns(report.getConfigWarns())
+          .setConsistencyCheckStatus(startupConsistencyCheck.getStatus())
+          .setDebug(getBoolean(PropertyKey.DEBUG)).setDiskCapacity(diskCapacity)
+          .setDiskCapacity(totalSpace).setDiskFreeCapacity(diskFreeCapacity)
+          .setDiskUsedCapacity(diskUsedCapacity).setFreeCapacity(FormatUtils
+              .getSizeFromBytes(mBlockMaster.getCapacityBytes() - mBlockMaster.getUsedBytes()))
+          .setInconsistentPathItems(startupConsistencyCheck.getInconsistentUris())
+          .setLiveWorkerNodes(mBlockMaster.getWorkerCount())
+          .setMasterNodeAddress(mMasterProcess.getRpcAddress().toString())
+          .setStartTime(CommonUtils.convertMsToDate(mMetaMaster.getStartTimeMs()))
+          .setStorageTierInfos(infos).setUptime(CommonUtils
+              .convertMsToClockTime(System.currentTimeMillis() - mMetaMaster.getStartTimeMs()))
+          .setUsedCapacity(FormatUtils.getSizeFromBytes(mBlockMaster.getUsedBytes()))
+          .setVersion(RuntimeConstants.VERSION);
+    });
```

Comment:
It may be cleaner to set the fields immediately after calculating them instead of storing them in local variables first. Up to you if you think this way looks more readable.

Reply:
I agree on this, I'll make a note on it.

Reply:
done